### DISCUSSION TITLE
Is the smooth command still working?

### DISCUSSION THREAD

**Discussion by jhiscocks** [mtex-toolbox/mtex#717] [Ask Anything] [answered]
I might be missing something incredibly basic here, but I was trying to smooth and fill holes using the _mtexdata small_ test data, and it's not working with MTEX 5.5.2.  The commands I used are below.  They all work fine except the last one, which seems to do nothing.  I also tried the fill command with no luck or error messages.

```matlab
%calculate grains
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',10*degree);
%remove EBSD data associated with small grains
ebsd(grains(grains.grainSize<3)) = [];
%recalculate the grains to remove boundaries around small grains.
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd('indexed'),'angle',10*degree)
%smooth the boundaries
grains = smooth(grains,5);
F = meanFilter;
F.numNeighbours = 3;
ebsdS = smooth(ebsd,F,'fill',grains);
figure; plot(ebsdS); hold on; plot(grains.boundary,'linewidth',2)
```

**Comment by kilir** [comment]
Hi,
have a look at the effect of `grains` you supply to smooth. If you want to fill the areas which were occupied by the small grains, you won't have to compute grains again after removing those and prior to smoothing.
Does that help?
Cheers, Rüdiger

**Comment by jhiscocks** [comment, marked as answer]
Thank you, your comment pointed me to the problem, which was actually a syntax error.  The smooth command should have been

`ebsdS = smooth(ebsd('indexed'),F,'fill',grains);`
